src/GraphBLAS-sharp/Abstracts.fs

```diff
@@ -20,9 +20,9 @@ type Matrix<'a when 'a : struct and 'a : equality>(nrow: int, ncol: int) =
 
     abstract Mxm: Matrix<'a> -> Mask2D option -> Semiring<'a> -> Matrix<'a>
     abstract Mxv: Vector<'a> -> Mask1D option -> Semiring<'a> -> Vector<'a>
-    abstract EWiseAdd: Matrix<'a> -> Mask2D option -> Monoid<'a> -> Matrix<'a>
-    abstract EWiseMult: Matrix<'a> -> Mask2D option -> Monoid<'a> -> Matrix<'a>
-    abstract Apply: Mask1D option -> UnaryOp<'a, 'b> -> Matrix<'b>
+    abstract EWiseAdd: Matrix<'a> -> Mask2D option -> Semiring<'a> -> Matrix<'a>
```

Comment:
It is no convenient to use monoid for elementwise multiplication because in monoid definition binary operation is called "append". In addition it is added for simmetry

Reply:
Yes, the naming convention is slightly strange, but monid is a set with the operation and we need only this operation for elementwise operations. So, semantically we should use monid. Maybe we should rename ```append``` to ```operation```.

Reply:
Semantically, addition and multiplication are different. Multiplication does not imply the presence of a zero element, so using a monoid is incorrect, i think.

Reply:
You are right. I forgot, that zeroes, which we won't save during multiplication, depend on the "add" monoid in semiring, not "multiply".